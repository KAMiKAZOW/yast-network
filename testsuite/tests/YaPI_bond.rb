# encoding: utf-8

module Yast
  class YaPIBondClient < Client
    def main
      Yast.import "Testsuite"
      Yast.import "Assert"

      @READ = {
        "init"      => { "scripts" => { "exists" => false } },
        "target"    => {
          "size"        => 27,
          "string"      => "laptop.suse.cz",
          "bash_output" => "laptop.suse.cz"
        },
        "probe"     => { "architecture" => "i386" },
        "sysconfig" => {
          "console" => { "CONSOLE_ENCODING" => "UTF-8" },
          "network" => {
            "config" => {
              "NETCONFIG_DNS_STATIC_SERVERS"    => "208.67.222.222 208.67.220.220",
              "NETCONFIG_DNS_STATIC_SEARCHLIST" => "suse.cz suse.de"
            },
            "dhcp"   => {
              "DHCLIENT_SET_HOSTNAME"   => "no",
              "WRITE_HOSTNAME_TO_HOSTS" => "no"
            }
          }
        },
        "routes"    => [
          { "destination" => "default", "gateway" => "10.20.30.40" }
        ]
      }

      @EXEC = {
        "target" => {
          "bash_output" => {
            "exit"   => 0,
            "stdout" => "laptop.suse.cz",
            "stderr" => ""
          }
        }
      }

      # configuration in form as expected on YaPI external interface (e.g. how WY sends it).
      # FIXME: Note that currently YaPI expects some options even if they are not used (e.g. bond_option)
      @net_ifaces_config = {
        "interface" => {
          "bond0" => {
            "bootproto"   => "static",
            "ipaddr"      => "4.3.2.1/24",
            "bond"        => "yes",
            "bond_slaves" => "eth1 eth2 eth3",
            "bond_option" => ""
          }
        }
      }

      Testsuite.Init([@READ, {}, @EXEC], nil)

      Yast.import "YaPI::NETWORK"

      @write_succeeded = { "error" => "", "exit" => "0" }

      Assert.Equal(@write_succeeded, YaPI::NETWORK.Write(@net_ifaces_config))

      nil
    end
  end
end

Yast::YaPIBondClient.new.main
