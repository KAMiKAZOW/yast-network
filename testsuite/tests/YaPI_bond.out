Read	.target.tmpdir nil
Log	Failed to set temporary directory: nil
Dir	.network.section: []
Read	.target.stat "/sys/class/net/bond0/type" nil
Read	.target.string "/sys/class/net/bond0/type" "laptop.suse.cz"
Log	No such device: bond0
Read	.target.stat "/sys/class/net/bond0/type" nil
Read	.target.string "/sys/class/net/bond0/type" "laptop.suse.cz"
Read	.target.stat "/sys/class/net/bond0/type" nil
Read	.target.string "/sys/class/net/bond0/type" "laptop.suse.cz"
Write	.network.value."bond0".BONDING_MASTER "yes" true
Write	.network.value."bond0".BONDING_MODULE_OPTS "" true
Write	.network.value."bond0".BONDING_SLAVE0 "eth1" true
Write	.network.value."bond0".BONDING_SLAVE1 "eth2" true
Write	.network.value."bond0".BONDING_SLAVE2 "eth3" true
Write	.network.value."bond0".BOOTPROTO "static" true
Write	.network.value."bond0".IPADDR "4.3.2.1/24" true
Write	.network.value."bond0".STARTMODE "auto" true
Write	.network nil true
Execute	.target.bash_output " LANG=C TERM=dumb COLUMNS=1024 systemctl --no-legend --no-pager --no-ask-password show network.service  --property=Id  --property=MainPID  --property=Description  --property=LoadState  --property=ActiveState  --property=SubState  --property=UnitFileState  --property=FragmentPath " $["exit":0, "stderr":"", "stdout":"laptop.suse.cz"]
Execute	.target.bash_output " LANG=C TERM=dumb COLUMNS=1024 systemctl --no-legend --no-pager --no-ask-password stop network.service " $["exit":0, "stderr":"", "stdout":"laptop.suse.cz"]
Execute	.target.bash_output " LANG=C TERM=dumb COLUMNS=1024 systemctl --no-legend --no-pager --no-ask-password show network.service  --property=Id  --property=MainPID  --property=Description  --property=LoadState  --property=ActiveState  --property=SubState  --property=UnitFileState  --property=FragmentPath " $["exit":0, "stderr":"", "stdout":"laptop.suse.cz"]
Execute	.target.bash_output " LANG=C TERM=dumb COLUMNS=1024 systemctl --no-legend --no-pager --no-ask-password start network.service " $["exit":0, "stderr":"", "stdout":"laptop.suse.cz"]
Execute	.target.bash_output " LANG=C TERM=dumb COLUMNS=1024 systemctl --no-legend --no-pager --no-ask-password show network.service  --property=Id  --property=MainPID  --property=Description  --property=LoadState  --property=ActiveState  --property=SubState  --property=UnitFileState  --property=FragmentPath " $["exit":0, "stderr":"", "stdout":"laptop.suse.cz"]
